### PR TITLE
Only bootstrap or platform classloader can define package java or java.*

### PR DESCRIPTION
**Pull request by JasonFengJ9**
Only `bootstrap` or `platform classloader` can define package `java` or `java.*` 

Throw `j.l.LayerInstantiationException` if a `classloader` other than `bootstrap` or `platform classloader` attempts to define package `java` or `java.*`;
Modified `VM.initializeClassLoader API` to accept an integer `loaderType` to differentiate `bootstrap`, `platform`, and other `classloader`;
Assuming `platform classloader` is the second `classloader` initialized after `bootstrap classloader`.

Reviewer: @danheidinga

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

### CODE REVIEW

File: runtime/vm/vm_internal.h [hunk 1]
```diff
@@ -394,6 +394,14 @@ j9gs_deinitializeThread(struct J9VMThread *vmThread);
 UDATA initializeExclusiveAccess(J9JavaVM *vm);
 void shutDownExclusiveAccess(J9JavaVM *vm);
 
+/**
+ * BytecodeInterpreter.hpp
+ * FastJNI_com_ibm_oti_vm_VM.cpp
+ */
+#define CLASSLOADER_TYPE_OTHERS		0
+#define CLASSLOADER_TYPE_BOOT		1
+#define CLASSLOADER_TYPE_PLAGTFORM	2
```
Comment (gacholio) [inline runtime/vm/vm_internal.h]: Spelling.

**Comment by gacholio** [comment]
he constants should be defined as static finals in the VM class, and initialized by the C code (like many others are) to avoid the magic dependency between java and C.

**Comment by gacholio** [comment]
What happens to Java 8 with these changes?

**Comment by gacholio** [comment]
Rather than depending on the order of class loader creation, can you pass in the loader type when instantiating?

**Comment by JasonFengJ9** [comment]
Discussed with GAC, this PR doesn't affect `Java 8` (code in question are decorated by JPP flags), `Java 9` uses `JCL` `jdk.internal.loader.ClassLoaders.BOOT_LOADER & PLATFORM_LOADER`, existing `Java 9` code depends on first `classloader` creation is `BOOT_LOADER`, and this PR depends on the second one is `PLATFORM_LOADER`. 
Will make changes accordingly for others.

**Comment by JasonFengJ9** [comment]
@gacholio PR has been updated and ready for review.

**Comment by gacholio** [comment]
jenkins test sanity plinux

**Pull request merged**